### PR TITLE
LPS-180016, LPS-180017, LPS-180018, LPS-180019, LPS-180020, LPS-180021, LPS-180022, LPS-180023, LPS-180024, LPS-180025 Add test to get item based on document from asset library on site navigation menus page

### PR DESCRIPTION
**Pull request by Tim-Cao**
https://liferay.atlassian.net/browse/LPS-180016
https://liferay.atlassian.net/browse/LPS-180017
https://liferay.atlassian.net/browse/LPS-180018
https://liferay.atlassian.net/browse/LPS-180019
https://liferay.atlassian.net/browse/LPS-180020
https://liferay.atlassian.net/browse/LPS-180021
https://liferay.atlassian.net/browse/LPS-180022
https://liferay.atlassian.net/browse/LPS-180023
https://liferay.atlassian.net/browse/LPS-180024
https://liferay.atlassian.net/browse/LPS-180025

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by ruben-pulido** [comment]
ci:test:relevant

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java [hunk 1]
```diff
@@ -627,6 +627,75 @@ public void testGetSiteNavigationMenusPage() throws Exception {
...
+						LocaleUtil.getDefault(), DLFileEntry.class.getName())
+				).buildString());
+
+		page = navigationMenuResource.getSiteNavigationMenusPage(
+			testGroup.getGroupId(), Pagination.of(1, 10));
+
+		Assert.assertEquals(1, page.getTotalCount());
+		Assert.assertEquals(
+			"document",
+			page.fetchFirstItem(
```
Comment (ruben-pulido) [inline modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java:676]: Should we define a variable for `page.fetchFirstItem().getNavigationMenuItems()[0]` and order the asserts alphabetically?

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java [hunk 2]
```diff
@@ -25,9 +40,85 @@
...
+						LocaleUtil.getDefault(), JournalArticle.class.getName())
+				).buildString());
+
+		NavigationMenu getNavigationMenu =
+			navigationMenuResource.getNavigationMenu(
+				postNavigationMenu.getId());
+
+		Assert.assertEquals(
+			"structuredContent",
+			getNavigationMenu.getNavigationMenuItems()[0].getType());
```
[the remark this answers is not in the document]
Comment (Tim-Cao) [reply modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java]: Sure

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java [hunk 3]
```diff
@@ -231,7 +231,76 @@ public void testGetSiteNavigationMenusPage() throws Exception {
...
+				).buildString());
+
+		page = navigationMenuResource.getSiteNavigationMenusPage(
+			testGroup.getGroupId(), Pagination.of(1, 10));
+
+		Assert.assertEquals(1, page.getTotalCount());
+		Assert.assertEquals(
+			"document",
+			page.fetchFirstItem(
+			).getNavigationMenuItems()[0].getType());
```
[the remark this answers is not in the document]
Comment (Tim-Cao) [reply modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/NavigationMenuResourceTest.java:482]: Sure

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by ruben-pulido** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward:force

**Comment by ealonso** [comment]
ci:forward:force

**Pull request closed without merging**